tool/tsh/db.go

```diff
@@ -335,7 +335,13 @@ func onDatabaseConnect(cf *CLIConf) error {
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		opts = append(opts, WithLocalProxy(addr.Host(), addr.Port(0)))
+		host := addr.Host()
+		if database.Protocol == defaults.ProtocolPostgres {
+			// In case of psql connection the cli takes only domain names into consideration thus
+			// connecting to 127.0.0.1 will fail with "does not match host name "127.0.0.1" psql error.
+			host = "localhost"
+		}
```

Comment:
Instead of treating Postgres as a special case, can we just always connect to local proxy over `localhost` instead?

Reply:
This work for mongo and postgres but for mysql localhost is a special value and when mysql client get localhost is try to by default to use Unix domain socket for connection:
So for example: 
`mysql --user root --host localhost -p 3306`
will return a error:
```
Can't connect to local MySQL server through socket '/tmp/mysql.sock'
```
but when `127.0.0.1` address is used it works: 
`mysql --user root --host 127.0.0.1 -p 3306`

But we can bypass this by providing the "--protocol=TCP" flag